applications/dashboard/models/class.usermodel.php

```diff
@@ -1051,13 +1051,15 @@ public function joinUsers(&$data, $columns, $options = []) {
         }
 
         // Join the user data using prefixes (ex: 'Name' for 'InsertUserID' becomes 'InsertName')
-        $join = val('Join', $options, ['Name', 'Email', 'Photo']);
+        $join = ($options['Join'] ?? ['Name', 'Email', 'Photo']);
 
         foreach ($data2 as &$row) {
+            $isObj = is_object($row);
             foreach ($prefixes as $px) {
-                $iD = val($px.'UserID', $row);
+                $pxUserId = $px.'UserID';
+                $iD = $isObj ? ($row->$pxUserId ?? false) : ($row[$pxUserId] ?? false);
                 if (is_numeric($iD)) {
-                    $user = val($iD, $users, false);
+                    $user = ($users[$iD] ?? false);
```

Comment:
I have high confidence in this change not breaking anything because $users are fetched with `->getIDs` which has a strongly typed array return.